pkg/engine/planner/physical/planner.go

```diff
@@ -96,6 +108,7 @@ func (p *Planner) processMakeTable(lp *logical.MakeTable) ([]Node, error) {
 		node := &DataObjScan{
 			Location:  objects[i],
 			StreamIDs: streams[i],
+			Direction: p.state.direction, // apply direction from previously visited Sort node
```

Comment:
nit: should this be done in optimization step?

Reply:
I thought about this as well, but I concluded that applying a sort order is not an optimisation, but rather a functional requirement.

Reply:
that makes sense!